Replace parseurl with WHATWG URL API

Saturate opened the pull request:
More information here:

https://github.com/e18e/ecosystem-issues/issues/160

Saturate committed:
chore: Replace parseurl with WHATWG URL API

More information here:

https://github.com/e18e/ecosystem-issues/issues/160

eugef committed:
Merge branch 'master' into chore/replace-parseurl-dependency

eugef commented:
Hi @Saturate 

Copilot actually caught a real issue here.

*`this` isn’t a URL, it’s a request object, so you should use new `URL(this.url)`.

* Also, `this.url` might not always be an absolute URL. `parseurl` handled relative ones fine, but the URL constructor doesn’t. Adding a base URL (like Copilot suggested) will make sure relative URLs work correctly and avoid potential problems.

Example:
```
const req = {
    url: '/some/path?name=foo',
}

console.log(parse(req).pathname); // Output: '/some/path'
console.log(new URL(req.url).pathname); // TypeError: Invalid URL
```

Saturate committed:
Merge branch 'master' into chore/replace-parseurl-dependency

Saturate committed:
fix: use this.url with base URL in path getter

The URL constructor requires an absolute URL string, not a request
object. Use this.url with a constructed base URL to handle relative
paths like '/some/path?name=foo' that parseurl previously handled.

Also fix pre-existing eslint no-extraneous-dependencies error in
type declaration tests.

Saturate commented:
😅

eugef committed:
Merge branch 'master' into chore/replace-parseurl-dependency

eugef committed:
Merge branch 'master' into chore/replace-parseurl-dependency

eugef committed:
Merge branch 'master' into chore/replace-parseurl-dependency

eugef committed:
Merge branch 'master' into chore/replace-parseurl-dependency
